src/libcore/slice/mod.rs

```diff
@@ -2466,6 +2466,256 @@ impl<T> [T] {
     {
         self.is_sorted_by(|a, b| f(a).partial_cmp(&f(b)))
     }
+
+    /// Empty out a slice and return a pointer to its contents.
+    ///
+    /// This is equivalent to `mem::replace(&mut slice, &[])`.
+    #[inline]
+    #[unstable(feature = "slice_take", issue = "62280")]
+    fn take<'a>(slice: &mut &'a Self) -> &'a Self {
+        crate::mem::replace(slice, &[])
```

Comment:
```suggestion
        mem::take(slice)
```
(`&[T]`  implements `Default`)

Reply:
`mem::take(slice)` is so concise that there's probably no point in having `Self::take(slice)` anymore (I do realize that this method isn't public, so it doesn't really matter)

Reply:
Well it does provide nice symmetry with `take_mut`.  But that could be implemented using `mem::take`.